library/core/src/mem/drop_guard.rs

```diff
@@ -158,3 +224,28 @@ where
         fmt::Debug::fmt(&**self, f)
     }
 }
+
+#[unstable(feature = "drop_guard", issue = "144426")]
+pub struct UnitFn<F>(F);
```

Comment:
I don't understand how this works. If this is going to be public, at a minimum it should include docs explaining what it is for and how it works.